#ver=2.0
---
launch_node : "server" # Server: server
recordFilePath : "qosserver/qos/record.json"

# Settings for Params class under vehicle_interfaces/params.h
generic_prop:
    # Node name and namespace. The namespace will automatically adding prefix at node name, topic name, service name, etc..
    namespace : "V0"
    nodeName : "qosserver"

    # Device ID, the ID will automatically formed the name of nodes and publisher topics. For instance, 
    # the node name will be formed as "<nodeName>_<id>_node", and the publisher topic name will be formed as "<topic_name>_<id>". 
    # Note: If the topic use to be subscriptions, then the topic name will not be formed.
    id : 0

    # ################################################################################################
    # The rest of params will not be used. Change the following params under service.json instead.
    # ################################################################################################

    # Device information service configuration
    devInfoService : "devinfo_0" # Service name devinfo_0
    devInterface : "eth0" # eth0
    devMultiNode : True # False

    # QoS service configuration
    qosService : "qos_0" # Service name qos_0
    qosDirPath : "launch/qos"

    # Safety service configuration
    safetyService : "" # Service name safety_0

    # Time sync service configuration
    timesyncService : "timesync_0" # Service name timesync_0
    timesyncPeriod_ms : 600000.0 # 10 minute a sync for HPC. 2 minute a sync for sensors.
    timesyncAccuracy_ms : 2.0 # Restrict the duration of sync process time
    timesyncWaitService : False # Wait until time synced at construction.
